File: mooc/BankQueue.cpp
//
//  stlQueue.cpp
//  cppCode
//
//  Created by clockworkSong on 2020/2/9.
//  Copyright © 2020 clockworkSong. All rights reserved.
//

#include <iostream>
#include <queue>

using namespace std;

int main(int argc, const char * argv[]) {
    int N;
    cin >> N;
    queue<string> q;
    string op, id;
    cin >> op;
    while (op != "#") {
        if(op == "In")
        {
            cin >> id;
            if(q.size() == N)
                cout << id << " rejected." << endl;
            else{
                q.push(id);
                cout << id << " joined. Total:" << q.size() << endl;
            }
        }
        else{
            if(q.empty()){
                cout << "No one!" << endl;
            }
            else{
                id = q.front();
                q.pop();
                cout << id << " called. Total:" << q.size() << endl;
            }
        }
        cin >> op;
    }
    return 0;
}

File: mooc/BankQueue.py
N = int(input())
op = input().split()
# print(op)
q = []#人数
while op[0] != '#':
	if op[0] == 'In':
		pid = op[1]
		if len(q) == N:
			print(pid, 'rejected.')
		else:
			q.append(pid)
			print(pid, 'joined. Total:%d' % len(q))
	else:
		if len(q) == 0:
			print('No one!')
		else:
			pid = q.pop(0)
			print(pid, 'Called. Total:%d' % len(q))
	op = input().split()